prophy/utils.py

```diff
@@ -0,0 +1,56 @@
+import prophy
+from six import string_types
+
+
+def _is_namedtuple(object_):
+    if type(object_).__bases__ == (tuple,):
+        fields = getattr(type(object_), '_fields', None)
+        if isinstance(fields, tuple):
+            return all(isinstance(field_name, str) for field_name in fields)
+
+
+def _try_to_get_attr(obj, name):
+    try:
+        return getattr(obj, name)
+    except AttributeError as e:
+        return str(e)
+
+
+def jsonize(struct_, ordered=True):
```

Comment:
Did you give yaml a thought? It seems to be "best suited for configuration" [1], and this is rather configuration (schema/typesystem of messages) than data (actual message objects with values). It's a little lighter than json for human eyes, allows comments, but might be less popular in terms of available parsers, etc.
```
>>> print(json.dumps(data, indent=True))
{
 "foo": {
  "x": [
   1,
   2,
   3
  ],
  "y": "uhishfusd",
  "z": 1203
 }
}
>>> print(yaml.dump(data))
foo:
  x: [1, 2, 3]
  y: uhishfusd
  z: 1203
```
Can you point out disadvantages of yaml that I fail to see while writing this? :)

[1] https://www.json2yaml.com/yaml-vs-json

Reply:
Good point. `yaml` is much more readable than `json`. I think the reason to implement such exporter is that `JSON` can be used in Java Script out of the box - while `YML` needs some external libraries. I think exposing methods for both would be nice.

Reply:
Yaml is, and good to have here. But since it seems so easy to change json-> yaml (yaml.dump(json_struct)), do we need anything in our library? That can be done externally on the flow...

Reply:
No it's not needed. Even @aurzenligl did it in this way in his example. However there is one reason to make custom `yaml` frontend: hidden fields. Comments in yaml could help avoiding a confusion. E.g. a unpopulated dynamic array is not rendered in such a json, inactive optional field as well. A comment mentioning that fact could be very desirable.

Reply:
If function aims at preparing data structures for further serialization in json/yaml, name: `sanitize_*` could be used. It doesn't mention serialization itself nor serialization format, but it suggests conforming to some stricter standard.